core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -233,19 +235,21 @@ public void updateDataMapSchema(List<DataMapSchema> dataMapSchemaList,
    * @param dataMapProvider
    * @param dataMapSchema
    */
-  public synchronized void registerDataMapCatalog(DataMapProvider dataMapProvider,
+  public void registerDataMapCatalog(DataMapProvider dataMapProvider,
       DataMapSchema dataMapSchema) throws IOException {
-    initializeDataMapCatalogs(dataMapProvider);
-    String name = dataMapSchema.getProviderName();
-    DataMapCatalog dataMapCatalog = dataMapCatalogs.get(name);
-    if (dataMapCatalog == null) {
-      dataMapCatalog = dataMapProvider.createDataMapCatalog();
-      if (dataMapCatalog != null) {
-        dataMapCatalogs.put(name.toLowerCase(), dataMapCatalog);
+    synchronized (lockObject) {
+      initializeDataMapCatalogs(dataMapProvider);
+      String name = dataMapSchema.getProviderName().toLowerCase();
+      DataMapCatalog dataMapCatalog = dataMapCatalogs.get(name);
+      if (dataMapCatalog == null) {
+        dataMapCatalog = dataMapProvider.createDataMapCatalog();
+        if (dataMapCatalog != null) {
```

Comment:
here if dataMapCatlog is still null Line 246 then it will not register the data map schema is it intentional, can please add some comments

Reply:
okay